lrpar/src/lib/ctbuilder.rs

```diff
@@ -374,24 +383,59 @@ where
         }
 
         let inc = read_to_string(grmp).unwrap();
-        let grm = YaccGrammar::<StorageT>::new_with_storaget(yk, &inc)
-            .map_err(|errs| {
-                errs.iter()
-                    .map(|e| {
-                        let mut line_cache = NewlineCache::new();
-                        line_cache.feed(&inc);
-                        if let Some((line, column)) =
-                            line_cache.byte_to_line_num_and_col_num(&inc, e.spans()[0].start())
-                        {
-                            format!("{} at line {line} column {column}", e)
+        let ast_validation = ASTValidation::new(yk, &inc);
+        let warnings = ast_validation.ast().warnings().collect::<Vec<_>>();
+        let spanned_fmt = |x: &dyn Spanned, inc: &str, line_cache: &NewlineCache| {
+            if let Some((line, column)) =
+                line_cache.byte_to_line_num_and_col_num(inc, x.spans()[0].start())
+            {
+                format!("{} at line {line} column {column}", x)
+            } else {
+                format!("{}", x)
+            }
+        };
+
+        let res = YaccGrammar::<StorageT>::new_with_validation(yk, ast_validation);
+        let grm = match res {
+            Ok(grm) if self.warnings_are_errors.not() || warnings.is_empty() => {
```

Comment:
I haven't seen `not` before. Is this equivalent to `!self.warnings_are_errors`?

Reply:
[I actually wonder if the `if` clause might be more obvious as `if !(self.warnings_are_errors && !warnings.is_empty())` but it's perhaps a wash either way!]

Reply:
> I haven't seen `not` before. Is this equivalent to `!self.warnings_are_errors`?

Yeah it is totally equivalent, I just find the binding strength of `.not()` slightly more obvious than `!`, but not steadfastly resolved in any one way.

Reply:
I agree that `!` can be easily missed, but since it's more common (I just looked up `not` which says something :)), I'd prefer to go with that.

Reply:
I ended up swapping this if condition to the other case then using
 `self.warnings_are_errors && !warnings.is_empty()` there since it removed one layer of `!`